compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -63,6 +65,8 @@
     private Location currentPos;
     private Map<BConstantSymbol, BLangConstant> unresolvedConstants = new HashMap<>();
     private Map<String, BLangConstantValue> constantMap = new HashMap<String, BLangConstantValue>();
+    private ArrayList<BConstantSymbol> resolvingConstants = new ArrayList<>();
+    private HashSet<BConstantSymbol> cannotResolveConstants = new HashSet<>();
```

Comment:
```suggestion
    private HashSet<BConstantSymbol> unresolvableConstants = new HashSet<>();
```

Reply:
IMO, we should be able to do this analysis without another set field here, using a specific kind of `BLangConstantValue` to represent errors. I guess it's not a straightforward change atm since there may be other places relying on the value being null in error cases.